field/track.go

```diff
@@ -0,0 +1,332 @@
+package field
+
+import (
+	"errors"
+	"fmt"
+	"regexp"
+	"strings"
+	"time"
+)
+
+var _ Field = (*Track)(nil)
+
+type Track struct {
+	spec *Spec  `xml:"-" json:"-"`
+	data *Track `xml:"-" json:"-"`
+
+	// Available for Track 1
+	FixedLength bool `xml:"-" json:"-"`
+
+	// Available track number: 1,2,3
+	Number TrackNumber `xml:"number,omitempty" json:"number,omitempty"`
+
+	// Available for Track 1, Track 3
+	FormatCode string `xml:"FormatCode,omitempty" json:"format_code,omitempty"`
+
+	// Available for Track 1, Track 2, Track 3
+	PrimaryAccountNumber string `xml:"PrimaryAccountNumber,omitempty" json:"primary_account_number,omitempty"`
+
+	// Available for Track 1
+	Name string `xml:"Name,omitempty" json:"name,omitempty"`
+
+	// Available for Track 1, Track 2
+	ExpirationDate *time.Time `xml:"ExpirationDate,omitempty" json:"expiration_date,omitempty"`
+
+	// Available for Track 1, Track 2
+	ServiceCode string `xml:"ServiceCode,omitempty" json:"service_code,omitempty"`
+
+	// Available for Track 1, Track 2, Track 3
+	//  If track 3, the field describe security data + additional data
+	DiscretionaryData string `xml:"DiscretionaryData,omitempty" json:"discretionary_data,omitempty"`
+}
+
+type TrackNumber int
+
+func (v TrackNumber) Valid() bool {
+	if v == Track1 || v == Track2 || v == Track3 {
+		return true
+	}
+	return false
+}
+
+const (
+	Track1 TrackNumber = 1
```

Comment:
@mfdeveloper508 can we split Track into separate structs/fields: Track1, Track2 and Track3? It seems that it will simplify the code and make it easier to understand and maintain.